### PR TITLE
Update FV3-JEDI ctest case to match MPAS-JEDI case 

### PR DESCRIPTION
**Pull request by SamuelDegelia-NOAA**
## Description

This PR unifies the FV3-JEDI and MPAS-JEDI test cases in RDASApp. Now the FV3-JEDI test case and ctests are valid at the same time (2024-05-27, 00:00 UTC) which allows for better comparison between GSI (RRFSv1), FV3-JEDI, and MPAS-JEDI. Thanks to @chunhuazhou for providing the background files for this case! 

The case data are now linked in to `RDASApp/fix/expr_data/fv3_2024052700` through `init.sh` (similar to the MPAS-JEDI test data). The `link_fv3jedi_expr.sh` script is then added to create a new experiment directory during the build called `RDASApp/expr/fv3_2024052700`. New bumploc files and template jobs for running fv3-jedi and bump are also added for running this new case. The reference data are also updated for the new case.  

Staged data and tools are also added for running GSI-based EnVar for this test case. The GSI configuration is taken from the RRFS_A retros on WCOSS2 with some minor changes including disabling dBZ state variables, disabling the hybrid (now using pure EnVar), a different mesonetuselist, and turning off `ext_sonde`. 

Given that some developers are still using the old FV3-JEDI test for validation, the old case is still linked into `RDASApp/fix/rrfs-test-data/rrfs-data_fv3jedi_2022052619` and `RDASApp/bundle/rrfs-test-data/rrfs-data_fv3jedi_2022052619`. As such, the old test data can still be accessed for validation experiments using `RDASApp/scripts/setup_experiment.sh`. But only the new test data is linked into `RDASAppp/fix/expr_data` and used for the ctests. 

## Issue(s) addressed
No issue was opened for this PR. 

## Dependencies (if applicable)
None. 

## Checklist
- [x] I have performed a self-review of my own code.
- [x] I have run rrfs tests before creating the PR (if applicable).
- [x] I have staged the new FV3-LAM test data on `RDAS_DATA`
- [x] I have staged the GSI test data on `RDAS_DATA`

### CODE REVIEW

File: rrfs-test/scripts/setup_experiment.sh [hunk 1]
```diff
@@ -154,6 +154,7 @@ if [[ $GSI_TEST_DATA == "YES" ]]; then
   sed -i "s#@YOUR_PATH_TO_GSI@#${YOUR_PATH_TO_GSI}#g" ./run_gsi.sh
   sed -i "s#@SLURM_ACCOUNT@#${SLURM_ACCOUNT}#g"       ./run_gsi.sh
   sed -i "s#@MACHINE_ID@#${MACHINE_ID}#g"             ./run_gsi.sh
+  sed -i "s#ANAL_TIME@#2022052619#g"                  ./run_gsi.sh
```
Comment (guoqing-noaa) [inline rrfs-test/scripts/setup_experiment.sh:157]: Sorry for the picky comment, Lines 154-157 may be merged into one sed command.

**Review by guoqing-noaa — DISMISSED** [review]
LGTM.

We can run CI tests when it is ready.

File: rrfs-test/scripts/setup_experiment.sh (hunk 1 above)
Comment (guoqing-noaa) [reply rrfs-test/scripts/setup_experiment.sh:157]: I don't mean we have to modify this, but if we would like to modify it, here is a reference for running multiple sed commands in one line:
https://github.com/NOAA-EMC/rrfs-workflow/blob/62b7336628738c5d9f94a87a80aa864e15e0adc5/scripts/exrrfs_fcst.sh#L69

**Comment by SamuelDegelia-NOAA** [comment]
I just pushed an update to use the larger set of obs for the new ctests instead of the single ob yamls used previously. I had to remove the ATMS obs since FV3-JEDI expects fields with different names than our restart files have. I also had to recreate the domain check obs since the FV3 domain is slightly different from our MPAS domain. 

This commit also switches the FV3-JEDI ctest from LETKF to GETKF (split mode) to better match our MPAS-JEDI configuration.

**Review by ShunLiu-NOAA — APPROVED** [review]

**Comment by ShunLiu-NOAA** [comment]
Thank you @SamuelDegelia-NOAA

**Comment by SamuelDegelia-NOAA** [comment]
For reference, here is a comparison of the new FV3-JEDI domain with the current MPAS-JEDI domain. 

domain_comparison

**Pull request merged**